// contexts/currentTheme.js
import React, {createContext, useReducer} from 'react';

const initialState = {
    isLight: true,
    isAlert: true,
    heightMenu: 130,
    preloader: true,
    isModal: false
};

const reducer = (state, action) => {
    switch (action.type) {
        case 'SET_DARK_THEME':
            return {
                ...state,
                isLight: false
            };
        case 'SET_LIGHT_THEME':
            return {
                ...state,
                isLight: true
            };
        case 'SET_ALERT_ON':
            return {
                ...state,
                isAlert: true
            };
        case 'SET_ALERT_OFF':
            return {
                ...state,
                isAlert: false
            };
        case 'SET_HEIGHT_MENU':
            return {
                ...state,
                heightMenu: action.payload
            };
        case 'SET_PRELOADER':
            return {
                ...state,
                preloader: action.payload
            };
        case 'SET_MODAL':
            return {
                ...state,
                isModal: action.payload
            };
        default:
            return state
    }
};

export const CurrentThemeContext = createContext();

export const CurrentThemeProvider = ({children}) => {
    const value = useReducer(reducer, initialState);

    return (
        <CurrentThemeContext.Provider value={value}>
            {children}
        </CurrentThemeContext.Provider>
    )
};
